File: src/main/java/itmo/is/project/dto/user/spaceship/CreateSpaceshipRequest.java
package itmo.is.project.dto.user.spaceship;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.swagger.v3.oas.annotations.media.Schema;
import itmo.is.project.model.user.Spaceship;

public record CreateSpaceshipRequest(

        @Schema(example = "S")
        @JsonProperty(value = "size", required = true)
        Spaceship.Size size
) {
}

File: src/main/kotlin/itmo/isproject/dto/user/spaceship/CreateSpaceshipRequest.kt
package itmo.isproject.dto.user.spaceship

import com.fasterxml.jackson.annotation.JsonProperty
import io.swagger.v3.oas.annotations.media.Schema
import itmo.isproject.model.user.Spaceship

data class CreateSpaceshipRequest(

    @Schema(example = "S")
    @JsonProperty(value = "size", required = true)
    val size: Spaceship.Size
)
